multibody/kinematic/test/kinematic_evaluator_test.cc

```diff
@@ -0,0 +1,170 @@
+
+  // Test q = 0, legs straight down
+  VectorXd q(plant_->num_positions());
+  plant_->SetPositions(context.get(), q);
+  VectorXd v = Eigen::VectorXd::Constant(plant_->num_velocities(), 1);
+  plant_->SetVelocities(context.get(), v);
+
+  auto phi = evaluator.EvalFull(*context);
+  EXPECT_TRUE(CompareMatrices(Vector3d({0, 0, -1}), phi, tolerance));
+
+  auto phi_active = evaluator.EvalActive(*context);
+
+  VectorXd phi_active_expected(1);
+  phi_active_expected << -1;
+  EXPECT_TRUE(CompareMatrices(phi_active_expected, phi_active, tolerance));
+
+  auto J = evaluator.EvalFullJacobian(*context);
+  MatrixXd J_expected(3, 6);
+  J_expected.row(0) << 0, 0, 0, 0, 0, 0;
+  J_expected.row(1) << 1, 0, -1, -1, 0, -.5;
+  J_expected.row(2) << 0, 1, 0, 0, 0, 0;
+  EXPECT_TRUE(CompareMatrices(J, J_expected, tolerance));
+
+  auto J_active = evaluator.EvalActiveJacobian(*context);
+  MatrixXd J_active_expected(1, 6);
+  J_active_expected << 0, 1, 0, 0, 0, 0;
+  EXPECT_TRUE(CompareMatrices(J_active, J_active_expected, tolerance));
+
+  auto phidot = evaluator.EvalFullTimeDerivative(*context);
+  VectorXd phidot_expected(3);
+  phidot_expected << 0, -1.5, 1;
+  EXPECT_TRUE(CompareMatrices(phidot, phidot_expected, tolerance));
+
+  auto phidot_active = evaluator.EvalActiveTimeDerivative(*context);
+  VectorXd phidot_active_expected(1);
+  phidot_active_expected << 1;
+  EXPECT_TRUE(CompareMatrices(phidot_active, phidot_active_expected,
+      tolerance));
+
+  auto Jdotv = evaluator.EvalFullJacobianDotTimesV(*context);
+  VectorXd Jdotv_expected(3);
+  Jdotv_expected << 0, 0, 6.5;
+  EXPECT_TRUE(CompareMatrices(Jdotv, Jdotv_expected, tolerance));
+
+  auto Jdotv_active = evaluator.EvalActiveJacobianDotTimesV(*context);
+  VectorXd Jdotv_active_expected(1);
+  Jdotv_active_expected << 6.5;
+  EXPECT_TRUE(CompareMatrices(Jdotv_active, Jdotv_active_expected, tolerance));
+
+  // Non-default evaluator: non-vertical normal, non-zero offset
+  // Performing minimal tests
+  auto new_evaluator = WorldPointEvaluator<double>(*plant_, pt_A,
+      frame, Vector3d({1, 0, 0}), Vector3d({1, 2, 3}), true);
+
+  auto new_phi = new_evaluator.EvalFull(*context);
+  EXPECT_TRUE(CompareMatrices(Vector3d({-4, 2, -1}), new_phi, tolerance));
+  auto new_phi_active = new_evaluator.EvalActive(*context);
+  EXPECT_TRUE(CompareMatrices(Vector3d({-4, 2, -1}), new_phi_active,
+      tolerance));  
+}
+
+TEST_F(KinematicEvaluatorTest, DistanceEvaluatorTest) {
+  const double tolerance = 1e-6;
+
+  Vector3d pt_A({0, 0, -.5});
+  const auto& frame_A = plant_->GetFrameByName("right_lower_leg");
+  Vector3d pt_B({0, 0, -.5});
+  const auto& frame_B = plant_->GetFrameByName("left_lower_leg");
+
+  // Default evaluator: vertical normal, zero offset
+  auto evaluator = DistanceEvaluator<double>(*plant_, pt_A, frame_A, pt_B,
+      frame_B, .5);
+
+  auto context = plant_->CreateDefaultContext();
+
+  // Test q = pi/2, legs straight down
+  VectorXd q(plant_->num_positions());
```

Comment:
This probably doesn't affect the test result, but I think we want to initial q with all zeros